src/allmydata/util/tor_provider.py

```diff
@@ -98,33 +95,31 @@ def _failed(f):
 
 @inlineCallbacks
 def _launch_tor(reactor, tor_executable, private_dir, txtorcon):
+    """
+    Launches Tor, returns a corresponding ``(control endpoint string,
+    txtorcon.Tor instance)`` tuple.
+    """
     # TODO: handle default tor-executable
     # TODO: it might be a good idea to find exactly which Tor we used,
     # and record it's absolute path into tahoe.cfg . This would protect
     # us against one Tor being on $PATH at create-node time, but then a
     # different Tor being present at node startup. OTOH, maybe we don't
     # need to worry about it.
-    tor_config = txtorcon.TorConfig()
-    tor_config.DataDirectory = data_directory(private_dir)
 
     # unix-domain control socket
-    tor_config.ControlPort = "unix:" + os.path.join(private_dir, "tor.control")
-    tor_control_endpoint_desc = tor_config.ControlPort
+    tor_control_endpoint_desc = "unix:" + os.path.join(private_dir, "tor.control")
```

Comment:
I don't remember why we're doing this here, but (mostly for future reference) it would be ideal if we just let txtorcon do this (it'll also try `localhost:9051` and `localhost:9151` for default system-tor and default Tor Browser Bundle endpoints, IIRC).

Reply:
I filed https://tahoe-lafs.org/trac/tahoe-lafs/ticket/4033, please correct if my description is inaccurate.